dafny/lcf/def.dfy

```diff
@@ -60,34 +60,89 @@ function string_join(sep : string, parts : seq<string>) : string {
   else parts[0] + sep + string_join(sep, parts[1..])
 }
 
-datatype Builtin = NatLeq | NatGeq | NatNeq | SubString | SplitString | Length | Member | Reverse {
+function atom_strings(consts : seq<Const>) : Result<seq<string>> {
+  if forall i :: 0 <= i < |consts| ==> consts[i].Atom? then
+    Ok(seq(|consts|, i requires 0 <= i < |consts| => consts[i].val))
+  else Err
+}
+
+function lower_char(c : char) : char {
+  match (c)
+  case 'A' => 'a'
+  case 'B' => 'b'
+  case 'C' => 'c'
+  case 'D' => 'd'
+  case 'E' => 'e'
+  case 'F' => 'f'
+  case 'G' => 'g'
+  case 'H' => 'h'
+  case 'I' => 'i'
+  case 'J' => 'j'
+  case 'K' => 'k'
+  case 'L' => 'l'
+  case 'M' => 'm'
+  case 'N' => 'n'
+  case 'O' => 'o'
+  case 'P' => 'p'
+  case 'Q' => 'q'
+  case 'R' => 'r'
+  case 'S' => 's'
+  case 'T' => 't'
+  case 'U' => 'u'
+  case 'V' => 'v'
+  case 'W' => 'w'
+  case 'X' => 'x'
+  case 'Y' => 'y'
+  case 'Z' => 'z'
+  case _ => c
+}
```

Comment:
I think you may be able to rewrite this by converting to integer with `(c as int)`, doing the ascii math, and converting back with `(... as char)`.

https://dafny.org/dafny/DafnyRef/DafnyRef.html#sec-characters

Reply:
Done